uptane/clients/primary.py

```diff
@@ -441,6 +459,51 @@ def get_validated_target_info(self, target_filepath):
           'the wrong repository specified as the Director repository in the '
           'initialization of this primary object?')
+    
+    director_target = validated_target_info[self.director_repo_name]
+
+    for repository_name in validated_target_info.keys():
+      current_target = validated_target_info[repository_name]
+      
+      if 'custom' not in validated_target_info[repository_name]['fileinfo']:
+        raise uptane.Error('{} repo failed to include the custom field in a target. \nTarget metadata was: {}'.format(repository_name, repr(current_target)))
+
+      custom_target_metadata = validated_target_info[repository_name]['fileinfo']['custom']
+
+      for custom_field in ['hardware_id', 'release_counter']:   
+      
+        if custom_field not in custom_target_metadata:
+      
+          raise uptane.Error('{} repo failed to include the {} field in the custom field of the target. \nTarget metadata was: {}'.format(repository_name, custom_field, repr(current_target)))
+          # Using a set to keep track of the values of hardware ID and release counter. Since all repos should have the same value for release acounter and hardware ID, the set length should never be greater than 2. 
+        if repository_name == self.director_repo_name:
+          Director_HardwareID_Release_counter.add(custom_target_metadata[custom_field])
+          print("DIRECTOR", Director_HardwareID_Release_counter)
+        else:
+          Hardware_ID_Release_Counter.add(custom_target_metadata[custom_field])
+
+        if len(Hardware_ID_Release_Counter) > 2:
+          # Assuming that the director is the only repository that can be compromised
+          raise uptane.Error("Internal organization repos have mismatching values in a target with the field {}. The target is {}".format(custom_field, repr(current_target)))
+
+
+    if Hardware_ID_Release_Counter != Director_HardwareID_Release_counter:
+      
+      Bad_element_director = Director_HardwareID_Release_counter.difference(Hardware_ID_Release_Counter) 
+
+      if len(Bad_element_director) == 1:
+        if type(list(Bad_element_director)[0]) == str:
+          raise uptane.HardwareIDMismatch('Bad value for the field hardware_ID in the that did not correspond the value in the other repos. Value did not match between the director and the other repos. The value of director target is {}'.format(repr(director_target))) 
+        else:
+          raise uptane.BadReleaseCounterValue('Bad value for the field release_counter that did not correspond the value in the other repos. Value did not match between the director and the other repos. The value of director target is {}'.format(repr(director_target)))
+
+      elif len(Bad_element_director) == 2:
+        raise uptane.BadHardwareIDReleaseCounter("Both the values of hardwareID and release counters did not satisfy the requirements. The target was" + repr(current_target))
+
+
+
+
```

Comment:
I wanted to make a separate director repo and another one for all the local image repos (+ any other repo that has been authorized by the OEM and is kept safe). This way I can have the first check to see if the internal image repos are uncompromised. 
I use sets for this because of the unique element nature of the sets data structure, which would flag an error if the number exceeds 2, the number of fields we are checking for. 
The sets.difference() allows me to check if the director holds values which are the same as that of the other local/secure/offline imagerepos from the OEM.
And sets do all of this in constant time so I felt that this was the way to go. 

Do you still want me to change it?